video/out/vo_drm.c

```diff
@@ -301,7 +301,7 @@ static int reconfig(struct vo *vo, struct mp_image_params *params)
     };
 
     talloc_free(p->cur_frame);
-    p->cur_frame = mp_image_alloc(IMGFMT, p->device_w, p->device_h);
+    p->cur_frame = mp_image_alloc(IMGFMT, p->screen_w, p->screen_h);
     mp_image_params_guess_csp(&p->sws->dst);
     mp_image_set_params(p->cur_frame, &p->sws->dst);
```

Comment:
Unrelated to this PR, but: I'm just spotting that it doesn't check whether p->cur_frame is NULL (because mp_image_alloc can fail).